File: src/2908.minimum-sum-of-mountain-triplets-i.3176/solution.ts
function minimumSum (nums: number[]): number {
    const rightMins = new Array(nums.length);
    rightMins[rightMins.length - 1] = nums[nums.length - 1];
    for (let i = nums.length - 2; i > 1; i--) {
        rightMins[i] = Math.min(rightMins[i + 1], nums[i]);
    }
    let result = Infinity;
    let prev = nums[0];

    for (let i = 1; i < nums.length - 1; i++) {
        if (nums[i] > prev && nums[i] > rightMins[i + 1]) {
            result = Math.min(result, nums[i] + prev + rightMins[i + 1]);
        }
        prev = Math.min(prev, nums[i]);
    }
    return result === Infinity ? -1 : result;
}

File: src/2908.minimum-sum-of-mountain-triplets-i.3176/solution.java
class Solution {
    public int minimumSum(int[] nums) {
        int[] lefts = new int[nums.length];

        int min = nums[0];
        for(int i=1;i< nums.length-1;i++){
            lefts[i] = min;
            min = Math.min(min,nums[i]);
        }

        int result = Integer.MAX_VALUE;
        boolean found = false;
        min = nums[nums.length-1];
        for(int i=nums.length-2;i>0;i--){
            if(lefts[i]<nums[i] && min<nums[i] ){
                found = true;
                result = Math.min(result, nums[i]+lefts[i]+min   );
            }
            min = Math.min(min,nums[i]);
        }
        if(!found){
            return -1;
        }
        return result;
    }
}